### DISCUSSION TITLE
quarkus-redis-cache and opentelemetry

### DISCUSSION THREAD

**Discussion by akil-rails** [quarkusio/quarkus#33196] [Q&A]
The docs on [opentelemetry](https://quarkus.io/guides/opentelemetry) talk about JDBC drivers. Is there also support for quarkus-redis with open-telemetry? Especially when using AWS managed Redis with AWS X-ray (via opentelemetry)? Any suggestions on where to read up on the docs/code will be very useful.

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
Do you mean for tracing? 

For metrics, the redis client exposes key metrics.

**Reply by brunobat** [reply]
There is no Quarkus tracing instrumentation for the Redis Client.
We don't support OTel Metrics, yet.

**Reply by DSkoufis** [reply]
@brunobat is it in the roadmap to support tracing instrumentation for the Redis Client?

Also is there a way to enable a tracer ourselves? Looks like Vert.x Redis Client [supports tracing commands](https://vertx.io/docs/vertx-redis-client/java/#_tracing_commands) but I couldn't figure a way to add a custom tracer as in the [`io.quarkus.opentelemetry.runtime.tracing.intrumentation.InstrumentationRecorder#L44`](https://github.com/quarkusio/quarkus/blob/11c724f566946b1fa4550e019be24b34c93e4476/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/intrumentation/InstrumentationRecorder.java#L44)

**Reply by brunobat** [reply]
Created this https://github.com/quarkusio/quarkus/issues/39011
